ethcore/src/client/client.rs

```diff
@@ -490,22 +509,23 @@ impl Importer {
 	fn commit_block<B>(
 		&self,
 		block: B,
-		header: &Header,
+		// header: &Header,
```

Comment:
I guess that this change is performed, so we don't have to clone the header before calling `commit_block`?

Reply:
Exactly.

The flow of data in this are is pretty complex and I guess the whole PR is an attempt to explore if it's really necessary and see if it can be simplified. It seems to me that the places where data actually must be owned are fewer than first meets the eye. I don't expect big perf improvements tbh but beyond "cloning is bad" I think being clear about what is owned where makes the intention of the code clearer.

Reply:
agree, I just think this method is an attempt to separate the header from the block which obviously is bad.

IMHO, fewer function parameters reduce complexity and make the code more readable.